internal/controller/pipelinerollout_controller.go

```diff
@@ -236,3 +302,125 @@ func (r *PipelineRolloutReconciler) SetupWithManager(mgr ctrl.Manager) error {
 
 	return nil
 }
+
+func setPipelineDesiredStatus(obj *kubernetes.GenericObject, status string) (*kubernetes.GenericObject, error) {
+	unstruc, err := kubernetes.ObjectToUnstructured(obj)
+	if err != nil {
+		return nil, err
+	}
+
+	err = unstructured.SetNestedField(unstruc.Object, status, "spec", "lifecycle", "desiredPhase")
+	if err != nil {
+		return nil, err
+	}
+
+	newObj, err := kubernetes.UnstructuredToObject(unstruc)
+	if err != nil {
+		return nil, err
+	}
+	return newObj, nil
+}
+
+func isPipelinePausing(ctx context.Context, pipeline *kubernetes.GenericObject) bool {
+	numaLogger := logger.FromContext(ctx)
+	pipelineStatus, err := kubernetes.ParseStatus(pipeline)
+	if err != nil {
+		numaLogger.Errorf(err, "failed to parse Pipeline Status from pipeline CR: %+v, %v", pipeline, err)
+		return false
+	}
+
+	numaLogger.Debugf("pipeline status: %+v", pipelineStatus)
+
+	return numaflowv1.PipelinePhase(pipelineStatus.Phase) == numaflowv1.PipelinePhasePausing
+}
+
+func isPipelinePaused(ctx context.Context, pipeline *kubernetes.GenericObject) bool {
+	numaLogger := logger.FromContext(ctx)
+	pipelineStatus, err := kubernetes.ParseStatus(pipeline)
+	if err != nil {
+		numaLogger.Errorf(err, "failed to parse Pipeline Status from pipeline CR: %+v, %v", pipeline, err)
+		return false
+	}
+
+	numaLogger.Debugf("pipeline status: %+v", pipelineStatus)
+
+	return numaflowv1.PipelinePhase(pipelineStatus.Phase) == numaflowv1.PipelinePhasePaused
```

Comment:
could potentially take the two methods above and strip out the shared logic into a common function